webapp/Component.js

```diff
@@ -170,79 +191,104 @@ sap.ui.define([
-                            }
-                            /**
-                             * OPTIONALLY: transmit the SAP user context to Azure Application Insights
-                             *
-                             *   Consult the "managing personal data" guide Azure Application Insights to learn more about the implications of personal data processing.
-                             *       https://learn.microsoft.com/azure/azure-monitor/logs/personal-data-mgmt
-                             */
-                             window.appInsights.setAuthenticatedUserContext(that._UserID);
-                        }
+                    var request = {};
+                    var element = myInteractions[i];
+                    var date = new Date(element.start);
 
-                        //register subsequent Fiori app loadings (keep track of user navigations)
-                        sap.ushell.Container.getServiceAsync("AppLifeCycle").then(function(AppLifeCycleService) {         
-                            that.myAppLifeCycle = AppLifeCycleService;
-                            //act on each Fiori app load and bind "this" context by injecting "that"
-                            that.myAppLifeCycle.attachAppLoaded(that.handleFioriAppLoaded,that);
-                        });
-                    },
-                    function(error){
-                        console.log(error);
+                    // if there is more that one interaction, we use the application context data which is still present.
+                    // only on reaching
+                    if (i==(myInteractions.length - 1)){
+                        that.prepareSAPApplicationData(that, appName);
+                        that.payload.sapSemanticObject = that.sapSemanticObject || null;
                     }
-                );
+
+                    // session scoped data
+                    that.payload.SAPFioriUserID = that._UserID;
+                    that.payload.SAPFioriUserEmail = that._UserEmail;
+                    that.payload.SAPFioriUserFullName = that._UserFullName;
+
+                    // event scoped data
+                    that.payload.appID = that.appID || null;
+                    that.payload.appNameID = that.appID || null;
+                    that.payload.appID = that.appID || null;
+                    that.payload.applicationType = that.applicationType || null;
+                    that.payload.technicalAppComponentID = that.technicalAppComponentID || null; 
+                    that.payload.appSupportInfo = that.appSupportInfo || null;
+                    that.payload.appIntent = that.appIntent || null;
+                    that.payload.productName = that.productName || null;
+
+                    //supply standard duration field
+                    that.payload.duration = element.duration || null;
+                    //write custom start time
+                    request.time = date.toISOString();
```

Comment:
verification required if this design decision creates more clarity for the output